### PR TITLE
Remove addDependencyTo from parcel-bundler

### LINKED ISSUE #1920: Deprecation warning when using posthtml
# ❔ Question

When using the posthtml transforms via .posthtmlrc, I get a deprecation warning.

## 💻 Code Sample

.posthtmlrc
using posthtml-include 1.2.0

```
{
  "plugins": {
    "posthtml-include": {
      "root": "./src/library/"
    }
  }
}
```

will throw (multiple times):

```
⠋ Building...addDependencyTo is deprecated in favor of
result.messages.dependency; posthtml-loader >= v1.0.1 will
automatically add your imported files to webpack's file watcher.
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.7
| Node             | 10.7.0
| npm/Yarn         | 6.3.0 / 1.9.2
| Operating System | macOS 10.13.6

### PR DESCRIPTION
**Pull request by LandonSchropp**
# ↪️ Pull Request

Hi there! This issue was bugging me, so I thought I'd take a quick stab at fixing it. I'm not familiar with the Parcel codebase, and I might have missed something, so please let me know if this isn't the right way to go about this, or you're not interested in this fix at this time.

This fixes a persistent warning message when using `posthtml-include`. It's been about 10 months since `posthtml-loader` version `1.0.1` was released, and the warning makes it difficult to use Parcel.

Fixes #1920

## 💻 Examples

Here's an example of my project's build after adding and removing whitespace three times:

screen shot 2018-10-17 at 8 44 36 pm

## 🚨 Test instructions

The easiest way to test is to add an `<include name="template"></include>` to an HTML file.

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/parcel-bundler/src/transforms/posthtml.js [hunk 1]
```diff
@@ -38,15 +38,7 @@ async function getConfig(asset) {
   config = config || {};
   const plugins = config.plugins;
   if (typeof plugins === 'object') {
```
Comment (chocolateboy) [inline packages/core/parcel-bundler/src/transforms/posthtml.js]: Without `depConfig`, this conditional is redundant and should be removed.

Comment (LandonSchropp) [reply packages/core/parcel-bundler/src/transforms/posthtml.js]: Should I drop the condition and call `Object.keys(plugins).forEach(p => plugins[p]);` directly?

Comment (LandonSchropp) [reply packages/core/parcel-bundler/src/transforms/posthtml.js]: I suppose it would make sense to assume `plugins` is always an object.

Comment (LandonSchropp) [reply packages/core/parcel-bundler/src/transforms/posthtml.js]: Sorry for the delayed response. Just to clarify, I should pull out the condition and the iteration of the plugins completely, right?

Comment (chocolateboy) [reply packages/core/parcel-bundler/src/transforms/posthtml.js]: Yes.

**Comment by LandonSchropp** [comment]
@chocolateboy I wanted to give a gentle bump on this. Have you had a chance to look at the updates?

**Comment by chocolateboy** [comment]
@LandonSchropp Yes, it looks fine. I can't vouch for the the removed test, though, and GitHub doesn't appear to provide a way for me to resolve/remove my obsolete feedback.

**Comment by LandonSchropp** [comment]
@chocolateboy Any advice on what we need to do to get this guy merged? Is there anything else I should change? Should we ask somebody else to review for that test?

**Comment by LandonSchropp** [comment]
I'm closing this PR. It looks like it's not going to get merged.

**Comment by andrewsosa** [comment]
@chocolateboy Was there anything blocking this from being merged? Not seeing all these messages would be a huge QOL improvement.

**Pull request closed without merging**